extensions/pkg/controller/worker/genericactuator/machine_controller_manager.go

```diff
@@ -125,7 +125,12 @@ func (a *genericActuator) waitUntilMachineControllerManagerIsDeleted(ctx context
 
 func (a *genericActuator) scaleMachineControllerManager(ctx context.Context, logger logr.Logger, worker *extensionsv1alpha1.Worker, replicas int32) error {
 	logger.Info("Scaling machine-controller-manager", "replicas", replicas)
-	return kubernetes.ScaleDeployment(ctx, a.client, kutil.Key(worker.Namespace, McmDeploymentName), replicas)
+	err := kubernetes.ScaleDeployment(ctx, a.client, kutil.Key(worker.Namespace, McmDeploymentName), replicas)
+	if apierrors.IsNotFound(err) {
+		logger.Info("Machine-controller-manager is not deployed yet")
+		return nil
+	}
+	return err
```

Comment:
Hm, do we actually care about this? I'm asking because you handle this "not found" case explicitly and add a log message, but is it really relevant/does it help? Or can we simply go with

```suggestion
	return client.IgnoreNotFound(kubernetes.ScaleDeployment(ctx, a.client, kutil.Key(worker.Namespace, McmDeploymentName), replicas))
```

Reply:
I also do not think this log is necessary. Every time restore is called for the first (right after migration) time the MCM will not be available.